api/lib/infrastructure/utils/redis-rate-limiter.js

```diff
@@ -0,0 +1,21 @@
+const redis = require('redis');
+const { promisify } = require('util');
+const logger = require('../logger');
+const settings = require('../../config');
+
+class RedisRateLimiter {
+  constructor() {
+    if (settings.rateLimit.redisUrl) {
+      this._client = redis.createClient(settings.rateLimit.redisUrl);
+      this._client.scriptAsync = promisify(this._client.script).bind(this._client);
+      this._client.evalshaAsync = promisify(this._client.evalsha).bind(this._client);
+      this._client.on('error', (err) => logger.warn({ redisClient: 'rate-limit', err }, 'Error encountered'));
+      return this._client;
+    } else {
+      this._client = null;
+    }
+    return this._client;
+  }
+}
+
+module.exports = new RedisRateLimiter();
```

Comment:
Effectivement la creation d'une classe ne semble pas adapté, on pourrait faire par exemple:

```suggestion
const redis = require('redis');
const { promisify } = require('util');
const logger = require('../logger');
const settings = require('../../config');

const redisRateLimiter =  settings.rateLimit.redisUrl ? buildRedisRateLimiter() : null;

function buildRedisRateLimiter() {
      const rateLimiter = redis.createClient(settings.rateLimit.redisUrl);
      rateLimiter.scriptAsync = promisify(rateLimiter.script).bind(rateLimiter);
      rateLimiter.evalshaAsync = promisify(rateLimiter.evalsha).bind(rateLimiter);
      rateLimiter.on('error', (err) => logger.warn({ redisClient: 'rate-limit', err }, 'Error encountered'));
      return rateLimiter;
}

module.exports = redisRateLimiter;
```

Reply:
Ahhhhh l'appel à la fonction au chargement du fichier, laisser NodeJs gérer le singleton plutôt que V5
Je tente pour voir si ça marche